changelog/922.improvement.md

```diff
@@ -0,0 +1,31 @@
+## Problem
+Rasa knowledge base actions cannot infer the object type of an object directly from the user message. This prevents action_query_knowledge from providing a suitable response when a user asks for a certain attribute of an object unless the user first asks to list objects related to that object type even though the knowledge base has the relevant information. That is, the knowledge base actions require the slot `object_type` to be set to one of the primary key values in the knowledge base for it to search through the objects. Here is an example:
+```
+Your input ->  what is the price range of Berlin Burrito Company?
+Sorry, I'm not sure I understand. Can you rephrase?
+Your input ->  list some restaurants
+Found the following objects of type 'restaurant':
+1: Gong Gan
+2: I due forni
+3: Pfefferberg
+4: Lụa Restaurant
+5: Donath
+Your input ->  what is the price range of Berlin Burrito Company?
+'Berlin Burrito Company' has the value 'cheap' for attribute 'price-range'.
+```
+
+## Proposed solution
+- The improvement requires changes to the classes ActionQueryKnowledgeBase and InMemoryKnowledgeBase under rasa-sdk. 
+- The `object_type` can be inferred by  utilizing the entity classification (DIET) where object types are used as entities to annotate object names. 
```

Comment:
I think it should be "entity classification **model**". The word "model" is missing here